File: src/16_cost_of_internet/cost_of_internet.cpp
#include <iostream>
#include <algorithm>
#include <string>
#include <vector>

using namespace std;

const string DAYS[] = {"Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"};

int day_number(string s) {
    return int(find(DAYS, DAYS + 7, s) - DAYS);
}

int main(void) {
    int n;
    cin >> n;
    vector <int> tday(n + 1), thour(n + 1), tprice(n + 1);
    for (int i = 1; i <= n; ++i) {
        string day_s;
        cin >> day_s >> thour[i] >> tprice[i];
        tday[i] = day_number(day_s);
    }
    tprice[0] = tprice[n];
    int k, cost = 0;
    cin >> k;
    for (int i = 0; i < k; ++i) {
        string day_s;
        int hour, mb;
        cin >> day_s >> hour >> mb;
        int day = day_number(day_s);
        int j = 0;
        while (tday[j + 1] < day || tday[j + 1] == day && thour[j + 1] <= hour) ++j;
        cost += mb * tprice[j];
    }
    cout << cost << endl;
    return 0;
}

File: src/16_cost_of_internet/cost_of_internet.py
DAYS = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sut', 'San']

n = int(input())
costs = []
for i in range(n):
    day, hour, cost = input().split()
    day = DAYS.index(day)
    hour = int(hour)
    cost = int(cost)
    costs.append([day, hour, cost])

costs.insert(0, [0, 0, costs[-1][2]])

k = int(input())
result = 0
for i in range(k):
    day, hour, mbs = input().split()
    day = DAYS.index(day)
    hour = int(hour)
    mbs = int(mbs)
    for j in range(n + 1):
        if costs[j][0] > day:
            break
        if costs[j][0] == day and costs[j][1] > hour:
            break
    result += mbs * costs[j - 1][2]


print(result)


